### ISSUE TITLE
informational ticket: 7.06 regressions

### ISSUE THREAD

**Issue by karenetheridge** [Perl-Toolchain-Gang/ExtUtils-MakeMaker#235]
put links to known 7.06 regressions here.

**Comment by karenetheridge** [comment]
non-lax $VERSIONs are now rejected when checking if prereqs are installed:  https://rt.cpan.org/Ticket/Display.html?id=106873  -- and fixed here -- https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/pull/234

**Comment by karenetheridge** [comment]
11:24 <@rjbs> ExtUtils::MakeMaker 7.04, Unicode::GCString installs fine.
11:25 <@rjbs> With 7.06: https://gist.github.com/rjbs/e5a3566e4c747d69ddbe
11:31 <@rjbs> Worked in 7.05_04.  Does ont work in 7.05_05.
11:32 <@haarg> https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/commit/8e4059c6d6cc1bc9e12fdb98a9538e14de2ffd44
11:32 <+dipsy> [ xs_obj_opt method adds -o to all $(CCCMD) commands to put .o next to .c · Perl-Toolchain-Gang/ExtUtils-MakeMaker@8e4059c · GitHub ] 
11:33 <@haarg> pretty sure that's the commit that broke it
11:33 <@rjbs> certainly seems like it would be
11:39 <@BinGOs> that's the one that broke File::Libmagic too.

**Comment by karenetheridge** [comment]
https://rt.cpan.org/Ticket/Display.html?id=106843 -- Makefile format changed -- fix in the dist, not EUMM

**Comment by karenetheridge** [comment]
Changes in PASSTHRU -- many distribution failures documented here -- https://rt.cpan.org/Ticket/Display.html?id=106808

**Comment by Leont** [comment]
I'm in favor of a quick regression fix release, and leave new developments for 7.10

**Comment by ribasushi** [comment]
Another meta-regression, arguably with lesser priority as it affects dev workflows only, not full-on installations: https://rt.cpan.org/Ticket/Display.html?id=106927. Nevertheless filing it, as ramifications for build-systems relying on `-e blib` *might* be severe.

**Comment by Leont** [comment]
> With 7.06: https://gist.github.com/rjbs/e5a3566e4c747d69ddbe

Essentially, it's already trying to work around the lack of -o. This is all such a fucking mess! :-(

**Comment by andk** [comment]
I believe this isn't yet reported elsewhere: the commit c515847 broke the idiom

  make testdb TEST_FILE=t/02-blah.t

This used to create exactly a single debugger session on this single test file but since then calls a debugger session on the one-liner that calls harness. Known issue? Wrong place to report it? Need more details? Let me know when I should open a new ticket.

**Comment by haarg** [comment]
Issues with -o option filed as #237.

**Comment by haarg** [comment]
@andk please file a ticket for the issue.

**Comment by andk** [comment]
>>>>> On Tue, 08 Sep 2015 13:38:28 -0700, Graham Knop <notifications@github.com> said:

  > @andk please file a ticket for the issue.

Done: https://rt.cpan.org/Ticket/Display.html?id=106975
-- 
andreas

**Comment by mohawk2** [comment]
#237 is tested for, and fixed, on master, with https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/commit/35e0ee4920606bfe218b394b5ed3c5fb0a1f48cc.

**Comment by mohawk2** [comment]
Re @andk's issue, here is what I commented on the ticket:

> This was where I introduced some consistency, which changed things. Before the commit you mention, the Makefile looked like this:

```
test_dynamic :: dynamic
        PERL_DL_NONLAZY=1 $(FULLPERLRUN) "-MExtUtils::Command::MM" "-MTest::Harn
ess" "-e" "undef *Test::Harness::Switches; test_harness($(TEST_VERBOSE), '$(INST
_LIB)', '$(INST_ARCHLIB)')" $(TEST_FILES)

testdb_dynamic :: dynamic
        PERL_DL_NONLAZY=1 $(FULLPERLRUN) $(TESTDB_SW) "-I$(INST_LIB)" "-I$(INST_ARCHLIB)" $(TEST_FILE)
```

> After it, due to a refactoring for DRY, it looked like this:

```
test_dynamic :: dynamic subdirs-test_dynamic
        PERL_DL_NONLAZY=1 $(FULLPERLRUN) "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness($(TEST_VERBOSE), '$(INST_LIB)', '$(INST_ARCHLIB)')" $(TEST_FILES)

testdb_dynamic :: dynamic subdirs-test_dynamic
        PERL_DL_NONLAZY=1 $(FULLPERLRUN) $(TESTDB_SW) "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness($(TEST_VERBOSE), '$(INST_LIB)', '$(INST_ARCHLIB)')" $(TEST_FILES)
```

> You will note that in the historical version, testdb_dynamic ran against `TEST_FILE` (set to test.pl), while test_dynamic ran against `TEST_FILES`. In the refactored version, they both run against `TEST_FILES`.

> I would suggest this is an improvement. You can run tests against specified tests by giving `TEST_FILES` on the command line. What do you think?

**Comment by haarg** [comment]
@mohawk2 Think about what testdb actually does.  It runs the debugger against the test.  This is entirely pointless to run against the harness.  You've made it "consistent", but also consistently useless.

**Comment by mohawk2** [comment]
Specifying `TEST_FILES` vs `TEST_FILE` is consistently useless? I am too stupid to understand, please help me.

**Comment by andk** [comment]
>>>>> On Thu, 10 Sep 2015 16:46:03 -0700, mohawk2 <notifications@github.com> said:

  > Specifying TEST_FILES vs TEST_FILE is consistently useless? I am too stupid to
  > understand, please help me.

I did explain the issue in tiket
https://rt.cpan.org/Ticket/Display.html?id=106975

Please let's discuss it over there. If you do not understand my posting,
I'm willing to answer your questions. I'm not interested in discussing
how this is coded. If the idiom is working, I'm a happy user. I'm not
coding police. The idiom is not working anymore and as far as I can see
there is no replacement. This is why that ticket has "Broken" in the
subject line. Please try to address my concerns. And let me know what I
need to add to that ticket so that you understand what I am talking
about.

Thank you,
-- 
andreas

**Comment by Leont** [comment]
What if we'd use `TEST_FILE`, but make it default to `TEST_FILES`? Wouldn't that solve this issue?

**Comment by mohawk2** [comment]
Now @haarg put it the way he did, I understand the reasoning why it ought to be `TEST_FILE` (singular) for `testdb`: the way I now understand it, debugging more than one file isn't a meaningful thing.

I'll make a test (there is currently no testing at all for `testdb` behaviour) and a fix.

**Comment by Leont** [comment]
Excellent!

**Comment by mohawk2** [comment]
Test and fix in https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/commit/ace8b86c6557630a94c8a33d880d77a54c329ebc.

**Comment by andk** [comment]
>>>>> On Sat, 12 Sep 2015 07:24:46 -0700, mohawk2 <notifications@github.com> said:

  > Test and fix in ace8b86.

What a joy, thank you!
-- 
andreas

**Comment by mohawk2** [comment]
If all the issues mentioned here are in fact resolved, could it now be closed?
